core/src/main/kotlin/me/kcra/takenaka/core/mapping/util/mappings.kt

```diff
@@ -62,19 +62,25 @@ val MappingTreeView.ElementMappingView.hash: String
  * @param mask the modifier mask (you can get that from the [Modifier] class)
  * @return the modifier string, **may end with a space**
  */
-fun Int.formatModifiers(mask: Int = 0): String = buildString {
+fun Int.formatModifiers(mask: Int = 0, interfaceMethod: Boolean = false): String = buildString {
     val mMod = this@formatModifiers and mask
 
-    if ((mMod and Opcodes.ACC_PUBLIC) != 0) append("public ")
+    // remove public and abstract modifiers on interface methods, implicit in some cases
+    // see: https://docs.oracle.com/javase/specs/jls/se8/html/jls-9.html#jls-9.4
+    if (!interfaceMethod && (mMod and Opcodes.ACC_PUBLIC) != 0) append("public ")
     if ((mMod and Opcodes.ACC_PRIVATE) != 0) append("private ")
     if ((mMod and Opcodes.ACC_PROTECTED) != 0) append("protected ")
     if ((mMod and Opcodes.ACC_STATIC) != 0) append("static ")
+
+    if (interfaceMethod && (mMod and Opcodes.ACC_ABSTRACT) == 0 && (mMod and Opcodes.ACC_STATIC) == 0) {
```

Comment:
this logic does not account for the fact that interface methods can be [private in Java 9 and higher](https://docs.oracle.com/javase/specs/jls/se9/html/jls-9.html#jls-9.4)

> A method in the body of an interface may be declared public or private ([§6.6](https://docs.oracle.com/javase/specs/jls/se9/html/jls-6.html#jls-6.6)). If no access modifier is given, the method is implicitly public.

> Default methods are distinct from concrete methods ([§8.4.3.1](https://docs.oracle.com/javase/specs/jls/se9/html/jls-8.html#jls-8.4.3.1)), which are declared in classes, and from private interface methods, which are neither inherited nor overridden.

> **It is a compile-time error if an interface method declaration that contains the keyword private also contains the keyword abstract or default. It is permitted for an interface method declaration to contain both private and static.**